Sources/KlaviyoSwift/StateManagement.swift

```diff
@@ -210,28 +225,58 @@ struct KlaviyoReducer: ReducerProtocol {
  
         case .enqueueLegacyEvent(let legacyEvent):
             // TODO: Needs a few tests.
-            guard let apiKey = state.apiKey else {
+            guard case .initialized = state.initalizationState, let apiKey = state.apiKey else {
                 return .none
             }
-            // TODO: might need to update state based on data in here.
-            return .run { [state] send in
-                guard let request = try? legacyEvent.buildEventRequest(with: apiKey, from: state) else {
-                    return
-                }
-                await send(.enqueueRequest(request))
+            
+            guard let identifiers = legacyEvent.identifiers else {
+                return .none
             }
+            state.email = identifiers.email ?? state.email
+            state.anonymousId = identifiers.anonymousId ?? state.anonymousId
+            state.phoneNumber = identifiers.phoneNumber ?? state.phoneNumber
+            state.externalId = identifiers.externalId ?? state.externalId
+            guard let request = try? legacyEvent.buildEventRequest(with: apiKey, from: state) else {
+                return .none
+            }
+            state.enqueueRequest(request: request)
+            return .none
         case .enqueueLegacyProfile(let legacyProfile):
-            // TODO: Needs a few tests.
-            guard let apiKey = state.apiKey else {
+            guard case .initialized = state.initalizationState, let apiKey = state.apiKey else {
                 return .none
             }
-            // TODO: might need to update state based on data in here.
-            return .run { [state] send in
-                guard let request = try? legacyProfile.buildProfileRequest(with: apiKey, from: state) else {
-                    return
+            guard let identifiers = legacyProfile.identifiers else {
+                return .none
+            }
+            state.email = identifiers.email ?? state.email
+            state.anonymousId = identifiers.anonymousId ?? state.anonymousId
+            state.phoneNumber = identifiers.phoneNumber ?? state.phoneNumber
+            state.externalId = identifiers.externalId ?? state.externalId
```

Comment:
nit: but should we functionize this state setup somewhere? That's probably all we need but just thinking about if the amount of state we are sending to each request grows, we'll be maintaining that in two places currently

Reply:
I'm hoping this'll go away when we have a new api.

Reply:
But yah...I'm actually looking at testing this better now. Will do.

Reply:
Moved a mutating method on the state.